community/rust/alpine-target.patch

```diff
@@ -23,229 +23,99 @@ and distribution-specific quirks instead of polluting the main musl target of
-+        target_endian: "little".to_string(),
-+        target_pointer_width: "64".to_string(),
-+        target_c_int_width: "32".to_string(),
-+        data_layout: "e-m:e-i8:8:32-i16:16:32-i64:64-i128:128-n32:64-S128".to_string(),
-+        arch: "aarch64".to_string(),
-+        target_os: "linux".to_string(),
-+        target_env: "musl".to_string(),
-+        target_vendor: "alpine".to_string(),
-+        linker_flavor: LinkerFlavor::Gcc,
-+        options: TargetOptions {
-+            abi_blacklist: super::arm_base::abi_blacklist(),
-+            .. base
-+        },
-+    })
++    let mut base = super::armv7_unknown_linux_musleabihf::target()?;
++
++    base.llvm_target = "armv7-alpine-linux-musleabihf".to_string();
++    base.target_vendor = "alpine".to_string();
++    base.options.crt_static_default = false;
++    base.options.static_position_independent_executables = true;
++    base.options.need_rpath = true;
++
++    Ok(base)
 +}
 --- /dev/null
-+++ b/src/librustc_target/spec/armv6_alpine_linux_musleabihf.rs
-@@ -0,0 +1,41 @@
-+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
-+// file at the top-level directory of this distribution and at
-+// http://rust-lang.org/COPYRIGHT.
-+//
-+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
-+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
-+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
-+// option. This file may not be copied, modified, or distributed
-+// except according to those terms.
-+
-+use crate::spec::{LinkerFlavor, Target, TargetOptions, TargetResult};
++++ b/src/librustc_target/spec/i586_alpine_linux_musl.rs
+@@ -0,0 +1,14 @@
++use crate::spec::TargetResult;
 +
 +pub fn target() -> TargetResult {
-+    let mut base = super::linux_musl_base::opts();
-+
-+    // Most of these settings are copied from the arm_unknown_linux_gnueabihf
-+    // target.
-+    base.features = "+v6,+vfp2".to_string();
-+    base.max_atomic_width = Some(64);
-+    base.crt_static_default = false;
-+    base.static_position_independent_executables = true;
-+    base.need_rpath = true;
-+
-+    Ok(Target {
-+        llvm_target: "arm-alpine-linux-musleabihf".to_string(),
-+        target_endian: "little".to_string(),
-+        target_pointer_width: "32".to_string(),
-+        target_c_int_width: "32".to_string(),
-+        data_layout: "e-m:e-p:32:32-i64:64-v128:64:128-a:0:32-n32-S64".to_string(),
-+        arch: "arm".to_string(),
-+        target_os: "linux".to_string(),
-+        target_env: "musl".to_string(),
-+        target_vendor: "alpine".to_string(),
-+        linker_flavor: LinkerFlavor::Gcc,
-+        options: TargetOptions {
-+            abi_blacklist: super::arm_base::abi_blacklist(),
-+            target_mcount: "\u{1}mcount".to_string(),
-+            .. base
-+        },
-+    })
++    let mut base = super::i686_unknown_linux_musl::target()?;
++
++       base.otpions.cpu = "pentium4".to_string();
```

Comment:
```suggestion
+       base.options.cpu = "pentium4".to_string();
```

Reply:
Yup, couldn't push while I'm connected to the alpine infra vpn to correct this

Reply:
No problem. I pushed that and x86_64 built successfully.

Reply:
Yup, works for me on x86_64 too and aarch64. Currently compiling on aarch64 with our triplets again, since that didn't work before.